native-maven-plugin/src/main/java/org/graalvm/buildtools/maven/AbstractNativeImageMojo.java

```diff
@@ -243,7 +243,29 @@ protected List<String> getBuildArgs() throws MojoExecutionException {
 
         if (buildArgs != null && !buildArgs.isEmpty()) {
             for (String buildArg : buildArgs) {
-                cliArgs.addAll(Arrays.asList(buildArg.split("\\s+")));
+                if(buildArg.startsWith("\\Q") ||
+                        buildArg.startsWith("-H:ConfigurationFileDirectories")) {
+                    cliArgs.add(buildArg);
+                    continue;
+                }
+                String[] args = buildArg.split("\\s+");
+                int i=0;
+                while(i < args.length) {
+                    String a =args[i];
+                    if (a.charAt(0) == System.getProperty("user.home").charAt(0)) {
```

Comment:
Is this an attempt to detect `/` (or `\\`)? If so, please use `File.separatorChar`

Reply:
This is to detect if the argument is a path (e.g., /home/... or C:/...). If it is, then it might be split with the space regex buildArg.split("\\s+"). For example, if you have the argument `/home/My user/target/app-name.jar`, it will be split into `/home/My` and `user/target/app-name.jar` and added as separate CLI arguments, which will result in parsing errors

example in windows: 
```
buildArgs = ["-cp C:\\Users\\My User\\my-app\\target\\app-name.jar"]
// after splitting with spaces
with get cliArgs = ["-cp", "C:\\Users\\My", "User\\my-app\\target\\app-name.jar"]
// resulting in parsing issues 
```
so I think this is an issue in linux and also windows.